app/upgrades/types.go

```diff
@@ -0,0 +1,22 @@
+package upgrades
+
+import (
+	storetypes "cosmossdk.io/store/types"
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/pokt-network/poktroll/app/keepers"
+)
+
+// Upgrade represents a protocol upgrade in code. Once a `MsgSoftwareUpgrade` is submitted to the chain, and
+// `PlanName` matches the `Name` of the `Plan` inside the upgrade message, the upgrade will be scheduled for execution.
+type Upgrade struct {
```

Comment:
I like this structure but just curious if this is something we came up with or adopted from someone else?

Reply:
I've checked how [other networks](https://github.com/cosmos/chain-registry) perform upgrades, and the chains that have frequent upgrades do something similar. Some ([evmos, for example](https://github.com/evmos/evmos/blob/2cd4415bb705a26129524035368bbe6df665e43b/app/app.go#L1131)) prefer to hard-code upgrades, but clean up the code after the fact.